### PR TITLE
Add docker-compose config

### PR DESCRIPTION
**Pull request by adaszko**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -134,45 +134,20 @@ for an org registered under `acme.radicle.eth`. Replace `acme` with the name
...
-Make sure your TLS certificate files can be found under `$HOME/.radicle`. If you
-are using TLS termination, simply omit the `--tls-*` arguments.
-
-You may also want to detach the process (`-d`) and run with a TTY in interactive
-mode (`-it`).
+1. [Install docker-compose](https://docs.docker.com/compose/install/).
+1. `export DOCKER_TAG=$(git rev-parse HEAD)`
+1. `export SEED_USER_ID=$(id -u radicle-seed)` (not needed for building and pushing images, only for running)
+1. `docker-compose -f docker-compose.yml -f docker-compose.gcp.yml build`
+1. `docker-compose -f docker-compose.yml -f docker-compose.gcp.yml push`
```
Comment (cloudhead) [inline README.md:153]: I'd rather we leave the readme as-is and later add a section on docker-compose, as some people may want to build it themselves.

File: docker-compose.gcp.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  org-node:
+    logging:
+      driver: fluentd
+      options:
+        fluentd-address: 127.0.0.1:24224
+        tag: docker.org_node
+    volumes:
+      - /var/opt/radicle-seed/.radicle:/app/radicle
+    user: $SEED_USER_ID:$SEED_USER_ID
+    command: ["--subgraph", "https://gateway.thegraph.com/api/1758a78ae257ad4906f9c638e4a68c19/subgraphs/id/0x2f0963e77ca6ac0c2dad1bf4147b6b40e0dd8728-0", "--orgs", "0x8152237402e0f194176154c3a6ea1eb99b611482", "--rpc-url", "wss://eth-mainnet.alchemyapi.io/v2/RDOfd2g0-_-KEOh7LwBCcovkVRM2QI2m", "--log-format", "gcp"]
```
Comment (cloudhead) [inline docker-compose.gcp.yml:22]: The RPC URL should be taken out of here, as it contains a private api token.

File: docker-compose.gcp.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+version: "3.9"
+services:
+  http-api:
+    logging:
+      driver: fluentd
+      options:
+        fluentd-address: 127.0.0.1:24224
```
Comment (cloudhead) [inline docker-compose.gcp.yml:7]: What is fluentd? Is it always exposed on this port?

File: docker-compose.gcp.yml (hunk 2 above)
Comment (cloudhead) [reply docker-compose.gcp.yml:22]: On the CLI we use an env var called `ETH_RPC_URL`, maybe we can use the same here?

File: docker-compose.gcp.yml (hunk 3 above)
Comment (adaszko) [reply docker-compose.gcp.yml:7]: It's a GCP specific thing used for log gather.  That's why it's placed in a separate `docker-compose.gcp.yml` file that's used only on GCP.

File: docker-compose.gcp.yml (hunk 2 above)
Comment (adaszko) [reply docker-compose.gcp.yml]: OK, done.  I've put `ETH_RPC_URL` into `.env` in an encrypted form.  You may want to take a look at [sops](https://github.com/mozilla/sops#examples).  It very easy to use.  I've added the command to decrypt to the README file.

File: README.md (hunk 1 above)
Comment (adaszko) [reply README.md]: OK, reverted.

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    ports:
+      - 8776:8776/udp
+    volumes:
+      - $HOME/.radicle:/app/radicle
+    environment:
+      RUST_LOG: org-node=debug,info
+    init: true
+    container_name: org-node
+    restart: unless-stopped
+    command: ["--subgraph", "https://gateway.thegraph.com/api/1758a78ae257ad4906f9c638e4a68c19/subgraphs/id/0x2f0963e77ca6ac0c2dad1bf4147b6b40e0dd8728-0", "--orgs", "0x8152237402e0f194176154c3a6ea1eb99b611482", "--rpc-url", $ETH_RPC_URL]
```
Comment (cloudhead) [inline docker-compose.yml:31]: Should we leave the `--org` there? I guess it's an org related to this project, so it's not too bad, but idk.

File: org-node/src/lib.rs [hunk 5]
```diff
@@ -123,9 +124,11 @@ pub enum Error {
 }
 
 /// Run the Node.
-pub fn run(rt: tokio::runtime::Runtime, options: Options) -> Result<(), Error> {
+pub fn run(rt: tokio::runtime::Runtime, options: Options) -> anyhow::Result<()> {
     let paths = Paths::from_root(options.root).unwrap();
-    let identity = File::open(options.identity)?;
+    let identity_path = options.identity.clone();
+    let identity = File::open(options.identity)
+        .with_context(|| format!("Unable to open {:?}", &identity_path))?;
```
Comment (cloudhead) [inline org-node/src/lib.rs:131]:
```suggestion
        .with_context(|| format!("unable to open {:?}", &identity_path))?;
```

Comment (cloudhead) [reply org-node/src/lib.rs:131]: Will be concatinated with other error strings..

File: README.md [hunk 6]
```diff
@@ -176,3 +176,11 @@ are using TLS termination, simply omit the `--tls-*` arguments.
 
 You may also want to detach the process (`-d`) and run with a TTY in interactive
 mode (`-it`).
+
+### GCP Deployment
+
+In order to deploy on GCP, you will need to decrypt .env file that contains
```
Comment (cloudhead) [inline README.md:182]:
```suggestion
In order to deploy on GCP, you will need to decrypt the .env file that contains
```

File: README.md [hunk 7]
```diff
@@ -176,3 +176,11 @@ are using TLS termination, simply omit the `--tls-*` arguments.
...
+requisite configuration encrypted with
```
Comment (cloudhead) [inline README.md:183]:
```suggestion
additional required environment variables encrypted with
```

File: .env [hunk 8]
```diff
@@ -0,0 +1,9 @@
+DOCKER_TAG=latest
+ETH_RPC_URL=ENC[AES256_GCM,data:PCD1HqzgS+g7XxhsZ2dxsOrm6F9NfoS3vyheLqba/syLmsmHrLaiA4jOc2JdK348ws9jM2LAuSRq1O1kyuuYXa22zQ==,iv:aUkOvK9APf0kkv9Ls6ezfhU+GvBWWsyQrgrJOeyldOk=,tag:Vofdq8gq10KSkgFH1QNT8A==,type:str]
+sops_gcp_kms__list_0__map_resource_id=projects/radicle-services/locations/global/keyRings/sops/cryptoKeys/sops-key
+sops_gcp_kms__list_0__map_created_at=2021-09-07T14:22:16Z
+sops_mac=ENC[AES256_GCM,data:fgign6WX9t2RrWTomgiDPK4uofvk55h0kBUltP4kOqRR1jQoswnOWIt4IQJa5FkKcVcTZQA6pglhu3q25DOLfsFZIKWi8ljnfvFadOKuqFaGwf/gDuDr96m/tKBzTtp8yiRwWt/yBbYuEQ10154dTkMjhVwac0rGu1S0x6ikCng=,iv:X3H4f80bU2l7UOBVRKQumoD4qzwMDeZHIw4RV6bajJw=,tag:BUQPSjV5wIApng16ZQVtmg==,type:str]
+sops_lastmodified=2021-09-07T14:22:17Z
+sops_encrypted_regex=ETH_RPC_URL
+sops_gcp_kms__list_0__map_enc=CiQAw9HjBmWd2O5n4q9FpEf+NR1rMlYgUP4HPQsmW2GVDBh8EY8SSQCYOiXngEsdj+/hKtvAQ/gmak22KjJEz4ePtvyY7X3CZFT4ofGmZYkdeI5Bfwxhid3kISattILdxFf546hnXNZHwBMfpF01EVM=
+sops_version=3.7.1
```
Comment (cloudhead) [inline .env:9]: Interesting, though this will only work for us :thinking: - I wonder if it belongs here.

Comment (adaszko) [reply .env:9]: New keys (pgp, gcp kms, aws kms, age) can be added any time, so it isn't restricted only to people having access to that specific gcp project.

File: docker-compose.yml (hunk 4 above)
Comment (adaszko) [reply docker-compose.yml]: You're right that it's too specific for the default case.  Removed.

File: .env (hunk 8 above)
Comment (cloudhead) [reply .env:9]: Ok you'll have to explain!

**Review by cloudhead — APPROVED** [review]

**Pull request merged**